### PR TITLE
feat(hybridcloud) Add control generational caching table

### PR DESCRIPTION
**Pull request by markstory**
Add a control silo version of the cross-region generational caching model. I'd like to start high-frequency RPC calls coming from control to the regions.

### CODE REVIEW

File: src/sentry/services/hybrid_cloud/user/service.py [hunk 1]
```diff
@@ -195,12 +192,11 @@ def get_user_avatar(self, *, user_id: int) -> RpcAvatar | None:
...
 
 @back_with_silo_cache("user_service.get_user", SiloMode.REGION, RpcUser)
-def get_user(user_id: int) -> RpcUser:
+def get_user(user_id: int) -> RpcUser | None:
     users = user_service.get_many(filter=dict(user_ids=[user_id]))
     if len(users) > 0:
         return users[0]
-    else:
-        return RpcUser(is_anonymous=True)
+    return None
```
Comment (markstory) [inline src/sentry/services/hybrid_cloud/user/service.py:199]: Supporting optional values in cached service calls is a significant change, but will allow us to use caching in more places. Resources like subscriptions and integrations don't have simple nullobjects that we can use, and supporting `None` lets caching work well with the optional returns we have in many RPC services.

**Comment by azaslavsky** [comment]
Fixture + pysnap changes LGTM, that covers the OSPO review.

**Review by azaslavsky — APPROVED** [review]

File: src/sentry/services/hybrid_cloud/user/service.py (hunk 1 above)
Comment (corps) [reply src/sentry/services/hybrid_cloud/user/service.py:199]: 👍

Comment (corps) [reply src/sentry/services/hybrid_cloud/user/service.py:199]: I am generally ok with this -- that said, is there any chance we might consider a `NullWrapper` or sentinel?

Reason is to help distinguish "the data isn't there yet" from "there data is definitely None".  But it's a minor consideration.

**Review by corps — APPROVED** [review]

File: src/sentry/services/hybrid_cloud/user/service.py (hunk 1 above)
Comment (markstory) [reply src/sentry/services/hybrid_cloud/user/service.py:199]: Getting a NullObject that works with the typings could be fun. I'll see if I can get that working in a follow up change.

**Pull request merged**